### PR TITLE
Bugfix for locale_gen issue (#21624)

### PR DESCRIPTION
**Pull request by isbm**
Fixes the `locale-gen` issue, described in #21624 as well as adds a `verbose=True/False` option.

### CODE REVIEW

File: salt/modules/localemod.py [hunk 1]
```diff
@@ -248,24 +251,22 @@ def gen_locale(locale):
...
         valid = __salt__['file.search'](search, '^{0}$'.format(locale))
     else:  # directory-based search
-        search_parts = _split_locale(locale)
-        search_parts['codeset'] = ''
-        search_parts['charmap'] = ''
-        search_locale = _join_locale(search_parts)
-
         if on_suse:
-            search = '/usr/share/locales'
+            search = '/usr/share/locale'
```
Comment (isbm) [inline salt/modules/localemod.py]: This is another bug fix, BTW.

**Comment by jfindlay** [comment]
@isbm, excellent, thanks.

**Comment by jfindlay** [comment]
@isbm, since this is technically a bugfix, we can put it into 2015.2.  If you want you may resubmit this pull request against that branch or we can backport it, thanks.

**Comment by isbm** [comment]
@jfindlay I am backporting it right now.

**Pull request closed without merging**